examples/compose/vtcompose.go

```diff
@@ -0,0 +1,566 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	jsonpatch "github.com/evanphx/json-patch"
+	"io/ioutil"
+	"math"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/krishicks/yaml-patch"
+	"vitess.io/vitess/go/vt/log"
+)
+
+var (
+	tabletsUsed = 0
+	baseDockerComposeFile    = flag.String("base_yaml", "docker-compose.base.yml", "Starting docker-compose yaml")
+	baseVschemaFile    = flag.String("base_vschema", "base_vschema.json", "Starting vschema json")
+
+	topologyFlags = flag.String("topologyFlags",
+		"-topo_implementation consul -topo_global_server_address consul1:8500 -topo_global_root vitess/global",
+		"Vitess Topology Flags config")
+	webPort   = flag.String("webPort", "8080", "Web port to be used")
+	gRpcPort  = flag.String("gRpcPort", "15999", "gRPC port to be used")
+	mySqlPort = flag.String("mySqlPort", "15306", "mySql port to be used")
+	cell      = flag.String("cell", "test", "Vitess Cell name")
+	keyspaceData = flag.String("keyspaces", "test_keyspace:2:1:create_messages.sql,create_tokens.sql unsharded_keyspace:0:0:create_dinosaurs.sql,create_eggs.sql", "List of keyspace_name:num_of_shards:num_of_replica_tablets:schema_files:<optional>lookup_keyspace_name seperated by ' '")
```

Comment:
This is a good start. I think this metadata may better be covered by a yaml file, mainly for the sake of uniformity. It would be ideal if we can standardize on the expected fields in the file. I have a start here: https://github.com/vitessio/vitess/blob/master/examples/helm/101_initial_cluster.yaml, which is for the helm charts. We'll also have to look at the operator CRDs.